### PR TITLE
Bump github.com/go-openapi/* to newer versions

### PR DESCRIPTION
**Pull request by dims**
periodic bump. Here's what we are doing:
- Bump github.com/go-openapi/spec to 0.19.8 ( https://github.com/go-openapi/spec/compare/v0.19.3...v0.19.8 )
- Bump github.com/go-openapi/swag to v0.19.9 ( https://github.com/go-openapi/swag/compare/v0.19.5...v0.19.9 )

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/go.mod [hunk 1]
```diff
@@ -31,6 +33,9 @@ require (
 )
 
 replace (
+	github.com/asaskevich/govalidator => github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a
+	github.com/go-openapi/swag => github.com/go-openapi/swag v0.19.5
+	github.com/mitchellh/mapstructure => github.com/mitchellh/mapstructure v1.1.2
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/go.mod:38]: we can't have replace directives in published modules... this means these have to be bumped as well :-/

**Comment by liggitt** [comment]
update looks fine, need to bump transitive versions so lint-dependencies doesn't complain

**Comment by nikhita** [comment]
kustomize is the only component that requires go-openapi right now:

```console
$ go mod why github.com/go-openapi/spec
# github.com/go-openapi/spec
k8s.io/kubernetes/pkg/kubectl/cmd/convert
k8s.io/cli-runtime/pkg/resource
sigs.k8s.io/kustomize/api/krusty
sigs.k8s.io/kustomize/kyaml/openapi
github.com/go-openapi/spec
```

Dependency on go-openapi was eliminated in kustomize in https://github.com/kubernetes-sigs/kustomize/pull/3827 and a new release which includes this change was also created - https://github.com/kubernetes-sigs/kustomize/releases/tag/kyaml%2Fv0.10.18

Would it help if we bumped kustomize instead to remove go-openapi completely?

xref https://github.com/kubernetes/kubernetes/pull/101234#discussion_r617511838, https://github.com/kubernetes/kubernetes/pull/98946#discussion_r589798392, https://github.com/kubernetes-sigs/kustomize/issues/3705

**Comment by dims** [comment]
@nikhita possibly, `kustomize` traditionally has been tough to update, So will have to wait for other folks to help with it. I'd say we can land this and still land kustomize when they feel ready.

**Comment by liggitt** [comment]
> kustomize is the only component that requires go-openapi right now

we actually have an unexpressed dependency on `github.com/go-openapi/spec` because of generated openapi files as well

File: vendor/github.com/asaskevich/govalidator/patterns.go [hunk 2]
```diff
@@ -4,49 +4,51 @@ import "regexp"
...
+	CreditCard        string = "^(?:4[0-9]{12}(?:[0-9]{3})?|5[1-5][0-9]{14}|(222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12}|6(?:011|5[0-9][0-9])[0-9]{12}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|(?:2131|1800|35\\d{3})\\d{11}|6[27][0-9]{14})$"
```
Comment (liggitt) [inline vendor/github.com/asaskevich/govalidator/patterns.go:8]: @sttts we use this in vendor/k8s.io/kube-openapi/pkg/validation/strfmt/default.go and these regex changes modify our API validation, right?

File: vendor/github.com/asaskevich/govalidator/patterns.go [hunk 3]
```diff
@@ -4,49 +4,51 @@ import "regexp"
...
+	DNSName           string = `^([a-zA-Z0-9_]{1}[a-zA-Z0-9_-]{0,62}){1}(\.[a-zA-Z0-9_]{1}[a-zA-Z0-9_-]{0,62})*[\._]?$`
+	IP                string = `(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))`
+	URLSchema         string = `((ftp|tcp|udp|wss?|https?):\/\/)`
+	URLUsername       string = `(\S+(:\S*)?@)`
+	URLPath           string = `((\/|\?|#)[^\s]*)`
+	URLPort           string = `(:(\d{1,5}))`
+	URLIP             string = `([1-9]\d?|1\d\d|2[01]\d|22[0-3]|24\d|25[0-5])(\.(\d{1,2}|1\d\d|2[0-4]\d|25[0-5])){2}(?:\.([0-9]\d?|1\d\d|2[0-4]\d|25[0-5]))`
```
Comment (liggitt) [inline vendor/github.com/asaskevich/govalidator/patterns.go:39]: I don't see this one used in our validation

**Comment by liggitt** [comment]
I think https://github.com/kubernetes/kubernetes/pull/102098#discussion_r635252431 changes our API validation

**Comment by gautierdelorme** [comment]
@dims @nikhita I opened https://github.com/kubernetes-sigs/kustomize/pull/3827 to drop `go-openapi/spec,validate,strfmt` dependency from kustomize so we can remove it from k/k as well. `kyaml/v0.10.18`, `cmd/config/v0.9.11`, `api/v0.8.9` includes that change but I'm waiting on @monopole to release a new `kustomize/v4.x` version before bumping kustomize in k/k and dropping `go-openapi/spec,validate,strfmt` totally. Also, I expect that update to be pretty simple (similar to https://github.com/kubernetes/kubernetes/pull/101120).

@liggitt Yeah that was one of the reasons why we moved to https://github.com/kubernetes/kube-openapi for API validation. IMO we shouldn't bump those dependencies and just upgrade kustomize to drop `go-openapi/spec,validate,strfmt` entirely.

**Comment by liggitt** [comment]
closing this in favor of updating kustomize and dropping the dependency altogether sounds good to me

**Comment by gautierdelorme** [comment]
@dims @nikhita @liggitt I opened the PR at https://github.com/kubernetes/kubernetes/pull/102193.

**Comment by liggitt** [comment]
Thanks!  

/close 
In favor of https://github.com/kubernetes/kubernetes/pull/102193

**Pull request closed without merging**